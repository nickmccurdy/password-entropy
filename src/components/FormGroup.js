import Icon from '@fortawesome/react-fontawesome'
import camelCase from 'lodash/camelCase'
import PropTypes from 'prop-types'
import React from 'react'
import { FormGroup as BootstrapFormGroup, Label } from 'reactstrap'

export default function FormGroup({ children, label, icon }) {
  return (
    <BootstrapFormGroup>
      <Label for={camelCase(label)}><Icon icon={icon} /> {label}</Label>
      {children}
    </BootstrapFormGroup>
  )
}

FormGroup.propTypes = {
  children: PropTypes.node.isRequired,
  label: PropTypes.string.isRequired,
  icon: PropTypes.oneOfType(
    [PropTypes.object, PropTypes.array, PropTypes.string].map(type => type.isRequired)
  ).isRequired
}
