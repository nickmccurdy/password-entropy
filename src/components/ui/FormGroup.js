import Icon from './Icon'
import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import { Col, ControlLabel, FormGroup as BootstrapFormGroup } from 'react-bootstrap'

export default class FormGroup extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired,
    id: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired
  }

  render () {
    const { children, id, label, icon } = this.props

    return (
      <BootstrapFormGroup controlId={id}>
        <Col sm={2}>
          <ControlLabel><Icon name={icon}/> {label}</ControlLabel>
        </Col>
        <Col sm={10}>{children}</Col>
      </BootstrapFormGroup>
    )
  }
}
