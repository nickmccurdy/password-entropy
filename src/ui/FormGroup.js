import PropTypes from 'prop-types'
import React from 'react'
import { Col, ControlLabel, FormGroup as BootstrapFormGroup } from 'react-bootstrap'
import Icon from 'react-fa'

export default function FormGroup ({ children, id, label, icon }) {
  return (
    <BootstrapFormGroup controlId={id}>
      <Col sm={2}>
        <ControlLabel><Icon name={icon}/> {label}</ControlLabel>
      </Col>
      <Col sm={10}>{children}</Col>
    </BootstrapFormGroup>
  )
}

FormGroup.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired
}
