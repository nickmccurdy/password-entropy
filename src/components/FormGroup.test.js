import arrowsAltH from '@fortawesome/fontawesome-free-solid/faArrowsAltH'
import { shallow } from 'enzyme'
import React from 'react'

import FormGroup from './FormGroup'

test('FormGroup', () => {
  const wrapper = shallow(
    <FormGroup label="Length" icon={arrowsAltH}>Redacted</FormGroup>
  )

  expect(wrapper).toMatchSnapshot()
})
