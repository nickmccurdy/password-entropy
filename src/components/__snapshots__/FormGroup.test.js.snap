// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup 1`] = `
<FormGroup
  tag="div"
>
  <Label
    for="length"
    tag="label"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <FontAwesomeIcon$1
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f337",
            "M377.941 169.941V216H134.059v-46.059c0-21.382-25.851-32.09-40.971-16.971L7.029 239.029c-9.373 9.373-9.373 24.568 0 33.941l86.059 86.059c15.119 15.119 40.971 4.411 40.971-16.971V296h243.882v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.568 0-33.941l-86.059-86.059c-15.119-15.12-40.971-4.412-40.971 16.97z",
          ],
          "iconName": "arrows-alt-h",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
     
    Length
  </Label>
  Redacted
</FormGroup>
`;
